frontend/src/screens/apps/Index.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable react-refresh/only-export-components */
+import { useEffect, useState } from "react";
+import { useNavigate } from 'react-router-dom';
+
+import Loading from '../../components/Loading';
+import { App, NostrEvent, ListAppsResponse } from "../../types";
+
+function Connections() {
+  const [apps, setApps] = useState<App[] | null>(null);
+  const [eventsCounts, setEventsCounts] = useState<Record<App["id"], number> | null>(null);
+  const [lastEvents, setLastEvents] = useState<Record<App["id"], NostrEvent> | null>(null);
+  const navigate = useNavigate();
+
+  const handleRowClick = (appId: App["nostrPubkey"]) => {
+    navigate(`/apps/${appId}`);
+  };
+
+  useEffect(() => {
+    const fetchAppsData = async () => {
+      try {
+        const response = await fetch("/api/apps");
+        const data: ListAppsResponse = await response.json();
+        setApps(data.apps);
+        setEventsCounts(data.eventsCounts);
+        setLastEvents(data.lastEvent);
+      } catch (error) {
+        // TODO: Show error page
+        navigate("/apps?q=notfound");
```

Comment:
I don't think we should navigate to an error page. I would just log an error to the console for now